packages/compiler-core/src/transforms/vIf.ts

```diff
@@ -30,13 +30,31 @@ import { ErrorCodes, createCompilerError } from '../errors'
 import { processExpression } from './transformExpression'
 import { validateBrowserExpression } from '../validateExpression'
 import { CREATE_COMMENT, FRAGMENT } from '../runtimeHelpers'
-import { findDir, findProp, getMemoedVNodeCall, injectProp } from '../utils'
+import {
+  findDir,
+  findProp,
+  getMemoedVNodeCall,
+  injectProp,
+  isTemplateNode,
+} from '../utils'
 import { PatchFlagNames, PatchFlags } from '@vue/shared'
 
 export const transformIf = createStructuralDirectiveTransform(
   /^(if|else|else-if)$/,
   (node, dir, context) => {
     return processIf(node, dir, context, (ifNode, branch, isRoot) => {
+      const isTemplate = isTemplateNode(node)
+      if ((__DEV__ || !__BROWSER__) && isTemplate) {
+        const keyProp = findProp(node, `key`, false, true)
+        if (keyProp) {
+          context.onError(
```

Comment:
It shouldn't be thrown an error. I think I should use the user-passed key here, just like normal elements.
[see](https://play.vuejs.org/#eNqNkkFPwzAMhf+K8aUgje0Ap2mbBGgScAAESFxymVq3ZEuTKEnLUNX/jptqhQlUcav9Pj89u2nwytppXRHOceFTJ20AT6GyK6FlaY0L0ICjHFrInSkhYTQRWujUaB+g9AUsO/00uSWlDLwZp7KT5Ezoxay3YyMuApVWbQJxBTBUUJ/LfCkwuIoEwnxHn1yxqcDVYvafmWMsgpmsx30Z6C1/kYN45IoTDJ73zWUx3Xqj+VRNNy8wNaWVityjDZLvIXAOUem0DZ/j4z72OvPJoZ++U7r7o7/1+64n8MmRJ1dznEELG1dQ6OX1ywPt+XsQS5NViukR8Zm8UVWXsceuK51x7B9cTHsXf7jUxatf7wNpf1gqXofJNvIC+RHcjKz+HfdiehnnhG6x/QIC7c10)